core/services/route_match/RouteMatchSpecificationManager.php

```diff
@@ -0,0 +1,146 @@
+     */
+    private $specifications_factory;
+
+
+    /**
+     * RouteMatchSpecificationManager constructor.
+     *
+     * @param RouteMatchSpecificationCollection $specifications
+     * @param RouteMatchSpecificationFactory    $specifications_factory
+     */
+    public function __construct(
+        RouteMatchSpecificationCollection $specifications,
+        RouteMatchSpecificationFactory $specifications_factory
+    ) {
+        $this->specifications = $specifications;
+        $this->specifications_factory = $specifications_factory;
+        add_action('AHEE__EE_System__loadRouteMatchSpecifications', array($this, 'initialize'));
+    }
+
+
+    /**
+     * Perform any early setup required for block editors to functions
+     *
+     * @return void
+     * @throws CollectionLoaderException
+     * @throws CollectionDetailsException
+     */
+    public function initialize()
+    {
+        $this->populateSpecificationCollection();
+    }
+
+
+    /**
+     * @return CollectionInterface|RouteMatchSpecificationInterface[]
+     * @throws CollectionLoaderException
+     * @throws CollectionDetailsException
+     * @since $VID:$
+     */
+    private function populateSpecificationCollection()
+    {
+        $loader = new CollectionLoader(
+            new CollectionDetails(
+            // collection name
+                RouteMatchSpecificationCollection::COLLECTION_NAME,
+                // collection interface
+                'EventEspresso\core\domain\entities\route_match\RouteMatchSpecificationInterface',
+                // FQCNs for classes to add (all classes within each namespace will be loaded)
+                apply_filters(
+                    'FHEE__EventEspresso_core_services_route_match_RouteMatchSpecificationManager__populateSpecificationCollection__collection_FQCNs',
+                    array(
+                        'EventEspresso\core\domain\entities\route_match\specifications\admin',
+                        // 'EventEspresso\core\domain\entities\route_match\specifications\public',
+                    )
+                ),
+                // filepaths to classes to add
+                array(),
+                // file mask to use if parsing folder for files to add
+                '',
+                // what to use as identifier for collection entities
+                // using CLASS NAME prevents duplicates (works like a singleton)
+                CollectionDetails::ID_CLASS_NAME
+            ),
+            $this->specifications,
+            null,
+            $this->specifications_factory
+        );
+        return $loader->getCollection();
+    }
+
+
+    /**
+     * Given the FQCN for a RouteMatchSpecification, will return true if the current request matches
+     *
+     * @param string $routeMatchSpecificationFqcn fully qualified class name
+     * @return bool
+     * @throws InvalidClassException
+     * @since $VID:$
+     */
+    public function currentRequestMatches($routeMatchSpecificationFqcn)
```

Comment:
I'd prefer if the name of this followed the pattern typically used for methods returning boolean types and was maybe something like `isCurrentRequestMatchingRoute`

Reply:
I think that convention primarily exists for methods that are **getters** for boolean properties, but is not so strict otherwise, and doesn't always have to mean the method is prefixed with `is` (ie: `has`, `should`, etc are also common), but there are also lots of examples of method/function names that don't follow that pattern because the word usage already makes it clear that they return a truthy value. For example: `class_exists()`, `SplObjectStorage::contains()`, etc. 

The more important underlying concept is that the method name is readable and this works best if the name can be used as both an assertion or a question, for example

```php
if ( class_exists( $fqcn ) ) {
// or
if ( $this->contains ) {
// or
if ( $this->isTrue ) {
```

what I don't like about `isCurrentRequestMatchingRoute` is that to me it only works as a question and not an assertion, and sounds weird when used in actual code, ie:

```php
if ( $this->isCurrentRequestMatchingRoute( $route ) ) {
```

I'll try to think if a better name that isn't awkward but still clearly sounds like an assertion/question

Reply:
I'm fine with any name that provides clarity when the method is implemented.  I personally don't think the name I suggested sounds weird in use (as per your example), the context _is_ a condition, i.e. written another way "If this is [the] current request matching [the] route then" (so the method name drops the "the" words.  I just think consistent usage of `is` prefix for these types of methods makes it immediately clear what their purpose is.  

Another suggestion that might work is `isRequestMatchingRoute`?

This is the last comment I'll make on this method name, I don't want to make another long thread out of this.